### DISCUSSION TITLE
Defer Stylesheet Loading

### DISCUSSION THREAD

**Discussion by Slakkan** [vercel/next.js#52685] [Help]
### Summary

I want to improve our lighthouse score, one of the things it complains about is that I should "Avoid chaining critical requests" and it provides [this link](https://developer.chrome.com/docs/lighthouse/performance/critical-request-chains/?utm_source=lighthouse&utm_medium=devtools)

How can I fix this on Next 13 with the new app directory?

![image](https://github.com/vercel/next.js/assets/48371558/e2d1e320-2de6-44b0-99a6-ec23d426c4f8)

**Comment by masterbater** [comment]
https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading

**Reply by Slakkan** [reply]
@masterbater I want to keep my components as server components. If I make them all client components I might as well use React alone.

I only would like to not preload the scss modules so from <link preload ... to <link.../>

**Reply by masterbater** [reply]
It seems you still dont know how client components work. Its Server Rendered same behavior with pre next 13 release or page directory. RSC is a new concept and it doesnt support what you want. 
You still get the benefit of SSR with client components. 
This is what get me frustrated when someone trying app router. They dont know how to differentiate the use case of client and server components. Im also like that at first but reading docs and building a real project with app router makes me understand things

**Reply by Slakkan** [reply]
@masterbater so client components also get rendered on the server and hydrated in the client?

Thanks for the heads up @masterbater , I didn't know that. 

I'll investigate further and come back to you if I have further questions. You can close this thread for now, thanks!

**Reply by masterbater** [reply]
Several references
https://mui.com/material-ui/guides/next-js-app-router/

[this explanation](https://github.com/reactwg/server-components/discussions/4)

https://nextjs.org/docs/getting-started/react-essentials#client-components

**Comment by icyJoseph** [comment]
By the way, is this the right one thing to put efforts into? The lighthouse UI, or pagespeed, has a calculator, where you can see, which metric is the farthest away from 100, and then you can see which diagnostics to pay attention to. 

I think, it is pretty common to have main thread work, and LCP, with higher impact over a 100 score.

<img width="1385" alt="Screenshot 2023-07-17 at 11 01 51" src="https://github.com/vercel/next.js/assets/21013447/9e3e9f92-80a6-4695-83a2-22beb6465dfa">

**Comment by Slakkan** [comment]
@masterbater importing all my components with next/dynamic didn't solve the issue, and I have no problems of LCP. It seems there is no way of deferring my stylesheets with the current Nextjs version

**Reply by icyJoseph** [reply]
What other diagnostics could you improve upon? I am not sure that deferring critical CSS is a good idea, right?

**Reply by Slakkan** [reply]
@icyJoseph Nothing, everything else is 100%, this is the only issue lighthouse finds with my Nextjs application. It is really good still but if I can do it better I would like to :)

**Reply by icyJoseph** [reply]
Right, I just think, for example, if you defer your CSS maximally, so first you show your content, and then later, allow the CSS to complete, then the page will "style" up, and we might end up triggering Cumulative Layout Shift.

If we defer, we might see a flash of un-styled content, that might trigger more issues of its own.

I think, normally under 14kB of CSS for the "above the fold" of the screen is good. You have like 8kb for the entire thing? Not sure how much more you could improve that. Maybe if the second CSS resource is a CSS module, you could try to dynamically load that component?

**Reply by Slakkan** [reply]
@icyJoseph I tried using Critters with Nextjs to only load critical css but I hadn't had much success. If anyone can point me to a configuration that works I would immensely appreciate it.

**Reply by masterbater** [reply]
You dont need to obsess with optimization.

**Reply by rosaldo260** [reply]
Facing the same issue and didn't find any solution yet.

**Comment by hallatore** [comment]
It seems next bundles all css in the same file even if it comes from dynamically imported components.

Would be nice if next respects dynamic(...) and bundles that css link separately at the bottom of the page.
